cfpb/consumerfinance.gov
# Replace Hallo with Draftail in table blocks

The goal was to have replaced Hallo in rich text table blocks by the time Wagtail 2.8 was launched.
This change is necessary because rich text table cells no longer use Hallo for editing.

## Additions

Add support for Draftail in JobListingPageFormTests
- `cfgov/jobmanager/tests/models/test_pages.py`

Updates made to add Draftail support to the following files:
- `table-block.js`
- `package.json`
- `yarn.lock`
- `test.py`

## Removals

Hallo configuration to support rich text table cells is removed
Removed legacy wagtail admin rich text editor `Hallo` in `base.py`
Removed `hallotoolbar` from `table-block.css`
Updates made to remove Hallo support from the file:
- `table-block.css`
- `wagtail_hooks.py`


## Changes

- Converted from HTML to raw JS for Draftail initialization
- Updated modal close event listener when user clicks save button
- Changed `data-hallo-editor` to `richtext` in `insertRichTextDeleteControl`
- Replace `Hallo` with `Draftail` in table blocks in `cfgov/templates/wagtailadmin/js`
- Changed WAGTAILADMIN_RICH_TEXT_EDITORS to use DraftailRichText instead of HalloRichText in `test.py`

Renamed hallobutton to draftailButton in:
- html_editor.js
- ask_cfpb_tips.js


## How to test this PR

You can checkout this branch locally and use ./runserver.sh
or use the Docker container that this Pull Request creates to:
1. Open or create a wagtail page
2. Modify the rich text
3. Click to save the changes 
4. Verify the changes are correctly saved with no errors locked

## Screenshots


## Notes

I feel the priority is to verify the functionality of clicking the save button that closes the modal.
It is very important that whatever is in the rich text editor is written to the table cell.
If that feature is NOT working, the Pull Request should not be merged until that
operation is confirmed to be working once the save button is clicked by the user.

## todos

- Verify draftail is properly setup in the model.
- Verify rich text buttons for bold, italics, etc. are added in this PR


## Checklist



  - PR titles are used to generate the change log in [releases](../../releases); good ones make that easier to scan.
  - Consider prefixing, e.g., "Mega Menu: fix layout bug", or "Docs: Update Docker installation instructions".
  - [This repo’s docs](https://cfpb.github.io/cfgov-refresh/) (edit the files in the `/docs` folder) – for basic, close-to-the-code docs on working with this repo
  - CFGOV/platform wiki on GHE – for internal CFPB developer guidance
  - CFPB/hubcap wiki on GHE – for internal CFPB design and content guidance

### Front-end testing



#### Browser testing

Visually tested in the following supported browsers:



#### Accessibility


#### Other
